_source/_posts/2023-10-12-okta-documentation-chatbot.md

```diff
@@ -0,0 +1,172 @@
+---
+layout: blog_post
+title: "How to Build an Okta Documentation Chatbot in Python"
+author: tanish-kumar
+by: internal-contributor
+communities: [python]
+description: ""
+tags: []
+tweets:
+- ""
+- ""
+- ""
+image: blog/okta-documentation-chatbot/oktanaut.jpeg
+type: 
+---
+
+In today's fast-paced world of technology, developer support is essential to ensure a seamless experience for users navigating intricate platforms and APIs. As a Developer Support Intern at Okta, I embarked on a challenging yet rewarding endeavor to create a solution that would streamline accessing and comprehending the Okta developer documentation. The result of my efforts is ✨Oktanaut✨, a versatile Python chatbot designed to assist users in harnessing the power of Okta's developer documentation, available at developer.okta.com. 
+
+{% img blog/okta-documentation-chatbot/oktadevforumsdk.jpeg alt:"oktadevforumsdk" height:"400" %}{: .center-image } 
+– Image source: <cite>[Okta Developer GitHub](https://raw.githubusercontent.com/oktadev/.github/main/images/okta-dev-header.png)</cite> 
+
+## Navigating Uncharted Waters
+Learning to build a chatbot was a formidable challenge. I spent hours scouring Medium articles, sifting through Coursera courses, and experimenting with various libraries and APIs. Determined, I delved into the world of chatbot development, driven by a desire to create a valuable tool for the Okta community.
+
+## Evolution of a Python Chatbot
+My journey led me to develop two distinct versions of Oktanaut a Python chatbot. The first version harnessed the capabilities of the GPT-3.5 API, allowing it to generate responses based on a broad spectrum of user inquiries. This version could handle various questions, providing a dynamic interaction experience.
+
+{% img blog/okta-documentation-chatbot/api.jpeg alt:"api-image" height:"400" %}{: .center-image }
+– Image source: <cite>[OpenAI APIs with Python — Complete Guide](https://medium.com/@marc.bolle/openai-apis-with-python-complete-guide-d933fb770f95)</cite>
+
+The second version took a different approach. Using LLamaIndex, I crafted a custom chatbot exclusively trained on the Okta developer documentation. While this version's responses were more accurate due to its focused training, it needed to be more comprehensive in handling a diverse range of questions compared to the GPT-3.5-powered counterpart.
```

Comment:
Please make sure to link LlamaIndex, and is the capitalization of Okta developer documentation correct for our brand? (idk)

Reply:
so you are suggesting changing Okta developer documentation to Okta Developer Documentation?

Reply:
I think the suggestion is to check the style guides linked from https://docs.google.com/document/d/1GjvQQoJJLy4bJIS9Yk5vyzy9tnxDcKcMJ1-yqc-Z-f8/edit. https://github.com/okta/okta-developer-docs/wiki/Style-Guide demonstrates that we do capitalize Dev Docs.